onnxruntime/core/providers/nnapi/nnapi_builtin/builders/shaper.cc

```diff
@@ -397,6 +404,30 @@ Status Shaper::SqueezeImpl(const std::string& input_name,
   return Status::OK();
 }
 
+Status Shaper::DepthToSpaceImpl(const std::string& input_name,
+                                const int32_t blocksize,
+                                bool nchw,
+                                const std::string& output_name) {
+  const Shape& input_dimen = shape_map_.at(input_name);
+
+  // Make output dimensions
+  Shape output_dimen = shape_map_.at(input_name);
+  if (nchw) {
+    output_dimen[0] = input_dimen[0];
+    output_dimen[1] = input_dimen[1] / (blocksize * blocksize);
+    output_dimen[2] = input_dimen[2] * blocksize;
+    output_dimen[3] = input_dimen[3] * blocksize;
+  } else {  // nhwc
+    output_dimen[0] = input_dimen[0];
```

Comment:
does this block assume that the input/output do not have the same layout?

from the NNAPI docs for DEPTH_TO_SPACE input 2 it sounds like they do have the same layout:
"2: An optional ANEURALNETWORKS_BOOL scalar, default to false. Set to true to specify NCHW data layout for input0 and output0. Available since NNAPI feature level 3."

Reply:
thanks for picking this up. the default layout is NHWC for input/output. will modify this along with using the optional bool scalar to specify NCHW format.